src/Util/HexString.hpp

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2016 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#ifndef HEX_STRING_HPP
+#define HEX_STRING_HPP
+
+#include <array>
+#include <string_view>
+#include <stdexcept>
+
+static unsigned char
+ParseHexDigit(const unsigned char c) {
+  if (c >= '0' && c <= '9')
+    return c - '0';
+
+  if (c >= 'A' && c <= 'F')
+    return c - 'A' + 10;
+
+  if (c >= 'a' && c <= 'f')
+    return c - 'a' + 10;
+
+  // Invalid character: parsing should fail
+  throw std::invalid_argument("Invalid hex character encountered.");
+}
+
+/**
+* Parses a string of hex data into a byte array.
+*
+* @param hex_str A string containing the data in hexadecimal form.
+* @return An array filled with the specified data.
+* @throws std::invalid_argument if parsing failed or the hex string has
+*         the wrong size for the array.
+*/
+template<std::size_t len>
+std::array<std::byte, len>
+ParseHexString(const std::string_view &hex_str)
```

Comment:
Don't pass `string_view` by reference. This adds overhead for dereferencing, and cannot be optimized away sometimes.